CONTRIBUTING.md

```diff
@@ -0,0 +1,142 @@
+This document explains the practical process and guidelines for contributing.
+
+While developing a Lightning implementation is an exciting project that spans many domains
+(cryptography, peer-to-peer networking, databases, etc), contributors must keep in mind that this
+represents real money and introducing bugs or security vulnerabilities can have far more dire
+consequences than in typical projects. In the world of cryptocurrencies, even the smallest bug in
+the wrong area can cost users a significant amount of money.
+
+If you're looking for somewhere to start contributing, check out the [good first issue](https://github.com/acinq/eclair/issues?q=is%3Aopen+is%3Aissue+label%3A"good+first+issue") list.
+
+Another way to start contributing is by adding tests or improving them.
+This will help you understand the different parts of the codebase and how they work together.
+
+## Communicating
+
+We recommend using our Gitter [developers channel](https://gitter.im/ACINQ/developers).
+Introducing yourself and explaining what you'd like to work on is always a good idea: you will get
+some pointers and feedback from experienced contributors. It will also ensure that you're not
+duplicating work that someone else is doing.
+
+We use Github issues only for, well, issues (mostly bugs that need to be investigated).
+You can also use Github issues for [feature requests](https://github.com/acinq/eclair/issues?q=is%3Aissue+label%3A"feature+request").
+
+## Recommended Reading
+
+- [Bitcoin Whitepaper](https://bitcoin.org/bitcoin.pdf)
+- [Lightning Network Whitepaper](https://lightning.network/lightning-network-paper.pdf)
+- [Deployable Lightning](https://github.com/ElementsProject/lightning/raw/master/doc/deployable-lightning.pdf)
+- [Understanding the Lightning Network](https://bitcoinmagazine.com/articles/understanding-the-lightning-network-part-building-a-bidirectional-payment-channel-1464710791)
+- [Lightning Network Specification](https://github.com/lightningnetwork/lightning-rfc)
+- [High Level Lightning Network Specification](https://medium.com/@rusty_lightning/the-bitcoin-lightning-spec-part-1-8-a7720fb1b4da)
+
+## Recommended Skillset
+
+Eclair uses [Scala](https://www.scala-lang.org/) and [Akka](https://akka.io/).
+Some experience with these technologies is required to contribute.
+There are a lot of good resources online to learn about them.
+
+## Contributor Workflow
+
+To contribute a patch, the workflow is as follows:
+
+1. [Fork repository](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) (only the first time)
+2. Create a topic branch
+3. Add commits
+4. Open a pull request
+
+### Pull Request Philosophy
+
+Pull requests should always be focused. For example, a pull request could add a feature, fix a bug,
+or refactor code; but not a mixture.
+Please also avoid super pull requests which attempt to do too much, are overly large, or overly
+complex as this makes review difficult.
```

Comment:
I would also add something like "Pull request should always include a clean, detailed description of what they fix/improve, why, and how. Even if you think that it is obvious, don't be shy and add explicit details and explanations."